NameLayerMain/src/vg/civcraft/mc/namelayer/GroupManager.java

```diff
@@ -309,11 +320,26 @@ public boolean hasAccess(String groupname, UUID player, PermissionType perm) {
 			NameLayerPlugin.getInstance().getLogger().log(Level.INFO, "hasAccess failed (access denied), could not find group " + groupname);
 			return false;
 		}
-		
 		GroupPermission perms = getPermissionforGroup(group);
-		PlayerType rank = group.getPlayerType(player);	
-		
-		return perms.isAccessible(rank, perm);
+		for(PlayerType rank : getRecursivePlayerTypes(group, player)) {
+			if (perms.hasPermission(rank, perm)) {
+				//player has right rank in the group itself or at least one super group
+				return true;
+			}
+		}		
+		return false;
+	}
+	
+	private List<PlayerType> getRecursivePlayerTypes(Group group, UUID player) {
```

Comment:
Only thing I really don't like about this is subgroup privilege escalation. Basically, if you're explicitly a `member` on group B which is a subgroup of group A, but you're explicitly a `mod` on group A, once they are linked you are now _effectively_ a `mod` on group B, even though your explicit privilege level on group B IMHO should dominate (e.g. still just a member).

In other words, only recurse if the person is _not_ a member of the group.

I know I went into these resolution styles in that super long missive, and clearly opinions will vary, just to my thinking always recursing up the stack means a lot of accidental and potential undesired privilege escalations.